src/util/Permissions.js

```diff
@@ -32,14 +32,28 @@ class Permissions {
 
   /**
    * Gets all given permissions that are missing from the bitfield.
-   * @param {PermissionResolvable[]} permissions Permissions to check for
+   * @param {PermissionResolvable[]|Permissions} permissions Permissions to check for
    * @param {boolean} [checkAdmin=true] Whether to allow the administrator permission to override
    * @returns {PermissionResolvable[]}
    */
   missing(permissions, checkAdmin = true) {
+    if (permissions instanceof Permissions) return this.missing(permissions.toArray());
     return permissions.filter(p => !this.has(p, checkAdmin));
   }
 
+  /**
+   * Returns this object as an {@link Array} of {@link PermissionResolvable}.
+   * @param {boolean} [checkAdmin=true] Whether to allow the administrator permission to override
+   * @returns {PermissionResolvable[]}
+   */
+  toArray(checkAdmin = true) {
+    let arr = [];
```

Comment:
`const arr = [];` you can still push into the array, so we should still prefer const.